torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -115,10 +145,10 @@ class AutoAugment(_AutoAugmentBase):
         "ShearX": (lambda num_bins, image_size: torch.linspace(0.0, 0.3, num_bins), True),
         "ShearY": (lambda num_bins, image_size: torch.linspace(0.0, 0.3, num_bins), True),
         "TranslateX": (lambda num_bins, image_size: torch.linspace(0.0, 150.0 / 331.0 * image_size[1], num_bins), True),
-        "TranslateY": (lambda num_bins, image_size: torch.linspace(0.0, 150.0 / 331.0 * image_size[1], num_bins), True),
+        "TranslateY": (lambda num_bins, image_size: torch.linspace(0.0, 150.0 / 331.0 * image_size[0], num_bins), True),
```

Comment:
Oh snap. How did we end up with the wrong value here? I guess we will need to do a very thorough check of every line of code.

Also if you check the main area, the `image_size[0]` is used for TranslateX. Did you flip the size tuple?

Reply:
> How did we end up with the wrong value here?

Most likely copy-pasting stuff around.

> I guess we will need to do a very thorough check of every line of code.

I'll do another pass.

> Also if you check the main area, the `image_size[0]` is used for TranslateX. Did you flip the size tuple?

Yes. `image_size` is always `(height, width)`. In our old implementation we are flip-flopping around with the convention and for auto augment we are using `(width, height)`. Thus, the index swap is correct here.

Reply:
As discussed there is a bug on the way we handle sizes. Even our methods sometimes return h,w and others w,h. Example:
https://github.com/pytorch/vision/blob/25e2ec0dac86a37d2e0083a3432ca3594f8185e4/torchvision/prototype/transforms/functional/_utils.py#L11

I think we should avoid estimating the image size from the tensor or the image attributes and instead rely to the get_image_size kernels everywhere so that at least we are consistent.